src/screens/CartScreen.tsx

```diff
@@ -0,0 +1,351 @@
+import * as React from 'react';
+import {
+  Image,
+  Button,
+  View,
+  StyleSheet,
+  Text,
+  ActivityIndicator,
+  FlatList,
+} from 'react-native';
+import {useSelector, useDispatch} from 'react-redux';
+import * as Sentry from '@sentry/react-native';
+import Toast from 'react-native-toast-message';
+import LinearGradient from 'react-native-linear-gradient';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {RootState, AppDispatch} from '../reduxApp';
+import {selectImage} from './ToolStore';
+
+interface CartData {
+  sku: string;
+  name: string;
+  image: string;
+  id: number;
+  type: string;
+  price: number;
+  quantity: number;
+}
+interface subTotal {
+  quantity: number;
+  total: number;
+}
+export type UIToast = typeof Toast;
+
+const CartScreen = (props) => {
+  const dispatch = useDispatch();
+  const cartData = useSelector((state: RootState) => state.cart);
+  const [orderStatusUI, setOrderStatusUI] = React.useState(false);
+
+  const cartItems: Array<CartData> | [] = Object.values(cartData);
+  const computeCartTotal = (cartItems: Array<CartData>): subTotal => {
+    let aggregate = cartItems.reduce(
+      (acc, item) => {
+        acc.quantity += item.quantity;
+        acc.total += item.price;
+        return acc;
+      },
+      {total: 0, quantity: 0},
+    );
+    return aggregate;
+  };
+  const subTotalDisplay = (props: subTotal): React.ReactElement => {
+    const {quantity: q, total: t} = props;
+    const multiple = q > 1 ? 's' : '';
+    return (
+      <Text
+        style={{
+          marginTop: 20,
+          marginBottom: 20,
+          fontSize: 18,
+          fontWeight: '600',
+        }}>{`Subtotal (${q} item${multiple}): $${(t / 1000).toFixed(2)}`}</Text>
+    );
+  };
+  const performCheckoutOnServer = async () => {
+    // ----------- Sentry Start Transaction ------------------------
+    let transaction = Sentry.startTransaction({name: 'checkout'});
+    Sentry.configureScope((scope) => scope.setSpan(transaction));
+    // -------------------------------------------------------------
+
+    let data = await placeOrder(Toast);
+    // ----------- Sentry Finish Transaction -----------------------
+    const span = transaction.startChild({
+      data,
+      op: 'task',
+      description: `processing shopping cart result`,
+    });
+
+    span.finish();
+    transaction.finish();
+
+    // -------------------------------------------------------------
+  };
+  const placeOrder = async (
+    uiToast: null | UIToast = null,
+  ): Promise<Response> => {
+    setOrderStatusUI(true);
+    const data = {cart: Object.values(cartData)};
+    let response = await fetch(
+      'https://dustinbailey-flask-m3uuizd7iq-uc.a.run.app/checkout',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          email: 'test@sentry.io',
+        },
+        body: JSON.stringify(data),
+      },
+    ).catch((err) => {
+      throw new Error(err);
```

Comment:
So this .catch block is not the intended 'catch block which you have below.
We have this going on in our sentry-demos/tracing demo app too.
Leave it as is because there's no better approach I can think of right now, and there will always be unknown ways in which the server side can provide an unexpected response.

In Tracing, the browser in use even made a difference sometimes in the response object.